src/server/goodid/utils.js

```diff
@@ -0,0 +1,66 @@
+import axios from 'axios'
+import { all as allCountries } from 'country-codes-list'
+
+import { substituteParams } from '../utils/axios'
+import { get, repeat, toUpper } from 'lodash'
+
+export class GoodIDUtils {
+  constructor(httpApi) {
+    const http = httpApi()
+    const { request, response } = http.interceptors
+
+    request.use(req => substituteParams(req))
+    response.use(({ data }) => data)
+    this.http = http
+  }
+
+  async getCountryCodeFromIPAddress(ip) {
+    const countryCode = await this.http
+      .get('https://get.geojs.io/v1/ip/country/:ip.json', { ip })
+      .then(response => get(response, 'country_3'))
+
+    if (!countryCode) {
+      throw new Error(`Failed to get country code from IP address '${ip}': response is empty or invalid`)
+    }
+
+    return countryCode
+  }
+
+  async getCountryCodeFromGeoLocation(latitude, longitude) {
+    const countryCode = await this.http
+      .get('https://nominatim.openstreetmap.org/reverse', {
+        format: 'jsonv2',
+        lat: latitude,
+        lon: longitude
+      })
+      .then(response => get(response, 'address.country_code'))
+      .then(toUpper)
+
+    if (!countryCode) {
+      throw new Error(
+        `Failed to get country code from coordinates '${latitude}:${longitude}': response is empty or invalid`
+      )
+    }
+
+    return countryCode
+  }
+
+  getCountryCodeFromMobile(phoneNumber) {
+    const trimmedNumber = phoneNumber.replace(/[^\d]/g, '')
```

Comment:
we should already be using google or some other phone library.
use that to parse the number and get country code

Reply:
Before adding new library i've researched over GoodServer and found nothing similar
I know we use such library at GoodDapp (which is a phone input React component)
Maybe we use something at worker. But not here. 

If I loss something and You know where such library/service is used at GoodServer - please link here

Reply:
please use a library for that, because country code parsing depands on the rest of the number
we usually use `google-libphonenumber`

Reply:
ok will look, is it this one https://www.npmjs.com/package/google-libphonenumber, right ?

Reply:
yes